### PR TITLE
actions: adding cachix

### PR DESCRIPTION
**Pull request by wiredhikari**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
         with:
           command: fmt
           args: --all -- --check
-
+
```
Comment (elsirion) [inline .github/workflows/rust.yml:47]: any particular reason for this?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -84,9 +84,13 @@ jobs:
...
+          nix_path: nixpkgs=channel:nixos-unstable
+      - uses: cachix/cachix-action@v10
         with:
-          nix_path: nixpkgs=channel:nixos-21.11
-      - name: Running nix-build
-        run: nix-build default.nix
+          name: mini
+          signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+      - run: nix-build default.nix
+      - run: nix-shell --run "echo OK"
```
Comment (elsirion) [inline .github/workflows/rust.yml:96]: Good idea to test the shell! Was wondering for a moment what it is for though, maybe you could echo "building nix-shell works"?

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -84,9 +84,13 @@ jobs:
     name: Nix-Build Tests
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: cachix/install-nix-action@v15
+      - uses: actions/checkout@v2.4.0
+      - uses: cachix/install-nix-action@v16
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
```
Comment (elsirion) [inline .github/workflows/rust.yml:90]: Did you try running it on stable? Would be preferable imo (most production systems will use that), or even better run both (although that could become annoying on breaking changes).

**Review by elsirion — COMMENTED** [review]
Minor nits

**Comment by elsirion** [comment]
1. Does this cache too much? Any data dirs maybe? I had to stop the test because the LN node couldn't open a channel because there already was one, going into an infinite loop.
2. Please squash (second commit is changing code the first already touched without a good reason)

**Comment by wiredhikari** [comment]
> 1. Does this cache too much? Any data dirs maybe? I had to stop the test because the LN node couldn't open a channel because there already was one, going into an infinite loop.
> 
> 2. Please squash (second commit is changing code the first already touched without a good reason)

It only does binary cache and built derivations. We won't need to build derivations twice.

**Comment by wiredhikari** [comment]
Have shifted to nixos-22.05 as recently nixos-21.11 got deprected. [Link](https://nixos.org/manual/nixos/stable/release-notes.html#sec-release-21.11)

**Review by elsirion — APPROVED** [review]
Changes look good, but please squash the commits :)

**Pull request merged**